packages/block-library/src/term-template/edit.js

```diff
@@ -0,0 +1,277 @@
+
+	const handleOnClick = () => {
+		setActiveBlockContextId( blockContextId );
+	};
+
+	const style = {
+		display: isHidden ? 'none' : undefined,
+	};
+
+	return (
+		<li
+			{ ...blockPreviewProps }
+			tabIndex={ 0 }
+			// eslint-disable-next-line jsx-a11y/no-noninteractive-element-to-interactive-role
+			role="button"
+			onClick={ handleOnClick }
+			onKeyPress={ handleOnClick }
+			style={ style }
+		>
+			{ termName }
+		</li>
+	);
+}
+
+const MemoizedTermTemplateBlockPreview = memo( TermTemplateBlockPreview );
+
+/**
+ * Builds a hierarchical tree structure from flat terms array.
+ *
+ * @param {Array} terms Array of term objects.
+ * @return {Array} Tree structure with parent/child relationships.
+ */
+function buildTermsTree( terms ) {
+	const termsById = {};
+	const rootTerms = [];
+
+	terms.forEach( ( term ) => {
+		termsById[ term.id ] = {
+			term,
+			children: [],
+		};
+	} );
+
+	terms.forEach( ( term ) => {
+		if ( term.parent && termsById[ term.parent ] ) {
+			termsById[ term.parent ].children.push( termsById[ term.id ] );
+		} else {
+			rootTerms.push( termsById[ term.id ] );
+		}
+	} );
+
+	return rootTerms;
+}
+
+/**
+ * Renders a single term node and its children recursively.
+ *
+ * @param {Object}   termNode   Term node with term object and children.
+ * @param {Function} renderTerm Function to render a single term.
+ * @return {JSX.Element} Rendered term node.
+ */
+function renderTermNode( termNode, renderTerm ) {
+	const children =
+		termNode.children.length > 0 ? (
+			<ul>
+				{ termNode.children.map( ( childNode ) =>
+					renderTermNode( childNode, renderTerm )
+				) }
+			</ul>
+		) : null;
+
+	return (
+		<>
+			{ renderTerm( termNode.term ) }
+			{ children }
+		</>
+	);
+}
+
+export default function TermTemplateEdit( {
+	clientId,
+	context: {
+		termQuery: {
+			taxonomy,
+			order,
+			orderBy,
+			hideEmpty,
+			hierarchical,
+			parent,
+		} = {},
+	},
+} ) {
+	const [ activeBlockContextId, setActiveBlockContextId ] = useState();
+
+	const queryArgs = {
+		order,
+		orderby: orderBy,
+		hide_empty: hideEmpty,
+	};
+
+	const { records: terms, isResolving } = useEntityRecords(
+		'taxonomy',
+		taxonomy,
+		queryArgs
+	);
+
+	// Filter to show only top-level terms if "Show only top-level terms" is enabled.
+	const filteredTerms = useMemo( () => {
+		if ( ! terms || parent !== 0 ) {
+			return terms;
+		}
+		return terms.filter( ( term ) => ! term.parent );
+	}, [ terms, parent ] );
+
+	const { blocks } = useSelect(
+		( select ) => ( {
+			blocks: select( blockEditorStore ).getBlocks( clientId ),
+		} ),
+		[ clientId ]
+	);
+
+	const blockContexts = useMemo(
+		() =>
+			filteredTerms?.map( ( term ) => ( {
+				taxonomy,
+				termId: term.id,
+				classList: `term-${ term.id }`,
+			} ) ),
+		[ filteredTerms, taxonomy ]
+	);
+
+	const blockProps = useBlockProps( {
+		className: clsx( 'wp-block-term-template' ),
+	} );
+
+	if ( isResolving ) {
+		return (
+			<p { ...blockProps }>
+				<Spinner />
```

Comment:
Is `Spinner` the best placeholder here? Could we use something that matches the format/layout of the block?

Reply:
Nice idea, thanks! I've added some grey placeholders in list items in https://github.com/WordPress/gutenberg/pull/70747/commits/0f82d3bcbc428ab12e60198b5cf99b1dd8e0e210:


https://github.com/user-attachments/assets/861df0ea-6456-471f-aacb-b110975708ac

I've used a similar animation to what's currently used in the Navigation and LinkControl blocks. It might need some tweaking but I prefer this to the spinner.